src/components/GroupForm/BulkUserManager/hooks/useUsers/index.js

```diff
@@ -0,0 +1,94 @@
+import { useDataQuery } from '@dhis2/app-runtime'
+import { useState, useEffect } from 'react'
+import { useDebounce } from 'use-debounce'
+import { useQueries } from './useQueries'
+import { useSet } from './useSet'
+
+const FILTER_DEBOUNCE_MS = 375
+
+export const useUsers = ({ groupId, mode }) => {
+    const queries = useQueries({ groupId })
+    const membersDataQuery = useDataQuery(queries.members, { lazy: true })
+    const nonMembersDataQuery = useDataQuery(queries.nonMembers, { lazy: true })
+    const [prevMembers, setPrevMembers] = useState()
+    const [prevNonMembers, setPrevNonMembers] = useState()
```

Comment:
Keep track of previous values of `members` and `nonMembers` in order to render them whilst the next/previous page of users is loading. Once https://github.com/dhis2/app-runtime/pull/1050 is merged, this logic can be eliminated.